### PR TITLE
[tari-script] Fix sqlite3 database migrations

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- Fixed "Cannot add a NOT NULL column with default value NULL"  errors with the `outputs` table introduced by the latest database migrations for `tari-script`. Reading up on this confirmed that in sqlite3 it is not allowed to add a `NOT NULL` column after the table has been created previously.

**Edits after initial review comments:**
- Date integrity: Changed `commitment` and `hash` default `NULL` columns in `outputs` table to `NOT NULL`.
 ~~**Note:** To change `tx_id` to be `NOT NULL` will require some refactoring and left for a follow-upPR.~~
- Superfluous constraints: Changed all `NULL DEFAULT NULL` columns to `NULL`  because `NULL` columns are `NULL` by default.

## Motivation and Context
A new wallet could not be started in Ubuntu or Windows due to the above error.

## How Has This Been Tested?
Running console wallets in Ubuntu and Windows.

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by stringhandler — CHANGES_REQUESTED** [review]
It is not correct to replace these with NULLs. This should be a new database on this branch, so I don't see the need for backwards compatibility

File: base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql [hunk 1]
```diff
@@ -1,12 +1,12 @@
 -- This migration is part of a testnet reset and should not be used on db's with existing old data in them
 -- thus this migration does not accommodate db's with existing rows.
 ALTER TABLE outputs
-    ADD COLUMN script BLOB NOT NULL;
+    ADD COLUMN script BLOB NULL DEFAULT NULL;
```
Comment (stringhandler) [inline base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql]: NULL columns are NULL by default, the `DEFAULT NULL` adds a constraint that does nothing

File: base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs [hunk 2]
```diff
@@ -254,7 +255,7 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
                     if OutputSql::find(&k.to_vec(), &(*conn)).is_ok() {
                         return Err(OutputManagerStorageError::DuplicateOutput);
                     }
-                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Spent, None);
+                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Spent, Some(OsRng.next_u64()));
```
Comment (stringhandler) [inline base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs]:
```suggestion
                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Spent, None);
```

File: base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs [hunk 3]
```diff
@@ -264,7 +265,7 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
                     if OutputSql::find(&k.to_vec(), &(*conn)).is_ok() {
                         return Err(OutputManagerStorageError::DuplicateOutput);
                     }
-                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Unspent, None);
+                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Unspent, Some(OsRng.next_u64()));
```
Comment (stringhandler) [inline base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs]:
```suggestion
                    let mut new_output = NewOutputSql::new(*o, OutputStatus::Unspent, None);
```

File: base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
-ALTER TABLE outputs
-    ADD COLUMN height INTEGER NOT NULL;
-ALTER TABLE outputs
-    ADD COLUMN script_private_key BLOB NOT NULL;
-ALTER TABLE outputs
-    ADD COLUMN script_offset_public_key BLOB NOT NULL;
\ No newline at end of file
+
+PRAGMA foreign_keys=off;
+DROP TABLE outputs;
```
Comment (philipr-za) [inline base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql:5]: I find it very unlikely that Mac has a less strict sqlite3 binary than windows and linux? Are you sure you were not just running this migration in a location with an existing db? The fact that this migrations fail if there is an existing db is a feature not a bug. It stops someone from using this testnet on an existing database.

Comment (hansieodendaal) [reply base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql:5]: Very certain there were no old databases present; tested this on three platforms as well.

Comment (philipr-za) [reply base_layer/wallet/migrations/2021-03-23-082938_update-outputs-for-tari-script/up.sql:5]: Wow, weird that it works on mac. Well in that case this approach is fine.

File: base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs (hunk 2 above)
Comment (hansieodendaal) [reply base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs]: Yes, need to revert these

File: base_layer/wallet_ffi/Cargo.toml [hunk 5]
```diff
@@ -14,6 +14,7 @@ tari_p2p = { version = "^0.8", path = "../p2p" }
 tari_wallet = { version = "^0.8", path = "../wallet", features = ["test_harness", "c_integration"]}
 tari_shutdown = { version = "^0.8", path = "../../infrastructure/shutdown" }
 tari_utilities = "^0.3"
+tari_key_manager = { version = "^0.8", path = "../key_manager" }
```
Comment (philipr-za) [inline base_layer/wallet_ffi/Cargo.toml:17]: Is this supposed to be in this PR?

**Review by stringhandler — APPROVED** [review]

File: base_layer/wallet_ffi/Cargo.toml (hunk 5 above)
Comment (hansieodendaal) [reply base_layer/wallet_ffi/Cargo.toml:17]: It is fixing a compile error due to recent merging of development into this branch; was necessary after I rebased.

**Pull request merged**